x-pack/legacy/plugins/infra/public/pages/infrastructure/metrics_explorer/index.tsx

```diff
@@ -41,6 +42,16 @@ export const MetricsExplorerPage = injectI18n(
       handleLoadMore,
     } = useMetricsExplorerState(source, derivedIndexPattern);
 
+    // Generate an index pattern if one doesn't exist the moment the app starts.
+    // This will prevent clicking on "Open in visualize" from redirecting to the
+    // index pattern page
+    useEffect(
```

Comment:
I don't really like putting this here since it's mixing presentational logic with background effects. This is the kind of thing I'd place in an "on initialize" Saga if we were using Redux-Saga for side effects, and it's less clear to me how to handle this with Epics as they only seem to respond to actions rather than having init procedures.

If there's a better way to do this please let me know.